File: hello-php/src/Logger.php
<?php

namespace AudioStreamClient;

class Logger
{
    private static bool $verboseEnabled = false;

    public static function setVerbose(bool $enabled): void
    {
        self::$verboseEnabled = $enabled;
    }

    public static function debug(string $message): void
    {
        if (self::$verboseEnabled) {
            self::log('debug', $message);
        }
    }

    public static function info(string $message): void
    {
        self::log('info', $message);
    }

    public static function warn(string $message): void
    {
        self::log('warn', $message);
    }

    public static function warning(string $message): void
    {
        self::warn($message);
    }

    public static function error(string $message): void
    {
        self::log('error', $message);
    }

    private static function log(string $level, string $message): void
    {
        $timestamp = date('Y-m-d H:i:s.') . substr(microtime(), 2, 3);
        echo "[$timestamp] [$level] $message\n";
    }
}

File: hello-swift/Sources/Logger.swift
import Foundation

/// Simple logging utility
public class Logger {
    private static let dateFormatter: DateFormatter = {
        let formatter = DateFormatter()
        formatter.dateFormat = "yyyy-MM-dd HH:mm:ss.SSS"
        return formatter
    }()
    
    nonisolated(unsafe) private static var verboseEnabled = false
    
    public static func setVerbose(_ enabled: Bool) {
        verboseEnabled = enabled
    }
    
    public static func debug(_ message: String) {
        if verboseEnabled {
            log(level: "debug", message: message)
        }
    }
    
    public static func info(_ message: String) {
        log(level: "info", message: message)
    }
    
    public static func warn(_ message: String) {
        log(level: "warn", message: message)
    }
    
    public static func error(_ message: String) {
        log(level: "error", message: message)
    }
    
    private static func log(level: String, message: String) {
        let timestamp = dateFormatter.string(from: Date())
        print("[\(timestamp)] [\(level)] \(message)")
    }
}
